lib/df-detection.c

```diff
@@ -0,0 +1,242 @@
+#include <linux/debugfs.h>
+#include <linux/df-detection.h>
+#include <linux/fs.h>
+#include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/random.h>
+#include <linux/sched.h>
+#include <linux/slab.h>
+#include <linux/stacktrace.h>
+#include <linux/sys.h>
+#include <linux/types.h>
+
+static int is_intersect(struct dfetch_address_range a,
+			struct dfetch_address_range b, void *kernel_addr)
+{
+	void *a_end = (void *)((char *)a.start_address + a.len);
+	void *b_end = (void *)((char *)b.start_address + b.len);
+	size_t len;
+	void *addr = kernel_addr;
+
+	if (a.start_address <= b.start_address && a_end > b.start_address) {
+		len = (char *)(a_end > b_end ? b_end : a_end) -
+		      (char *)b.start_address;
+	} else if (b.start_address <= a.start_address &&
+		   b_end > a.start_address) {
+		unsigned long diff =
+		    (char *)a.start_address - (char *)b.start_address;
+
+		len = (char *)(a_end > b_end ? b_end : a_end) -
+		      (char *)a.start_address;
+		addr = ((char *)kernel_addr + diff);
+	} else
+		return 0;
+	get_random_bytes(addr, len);
+	pr_err("Random injection in %px, Length %lu \n", addr, len);
```

Comment:
I think it's better to move this line into get_random_bytes().
At some point the injection logic may become more complicated, and it may happen conditionally. In that case we won't need to print anything.

Reply:
you mean moving printing line into get_random_bytes()? get_random_bytes() is a function declared in random.h so in this way we will modifying an existing function

Reply:
Ah, sorry, I didn't realize that. Please ignore this comment.